near-sdk/src/promise.rs

```diff
@@ -417,6 +417,18 @@ impl serde::Serialize for Promise {
     }
 }
 
+impl borsh::BorshSerialize for Promise {
+    fn serialize<W: Write>(&self, _writer: &mut W) -> Result<(), Error> {
+        *self.should_return.borrow_mut() = true;
```

Comment:
I agree, I'm just trying to make the least cost change to avoid breakage. This will definitely be a thing to keep in mind as evolving the API because this seems like it has the potential to be a footgun